eth_account/local.py

```diff
@@ -59,3 +59,9 @@ def signTransaction(self, transaction_dict):
 
     def __bytes__(self):
         return self.privateKey
+
+    def __hash__(self):
+        return hash(self.privateKey)
```

Comment:
Implementing `__hash__` implies that the object is immutable, so we need to make it harder to accidentally change the private key.

That could look something like: moving `privateKey` to `_privateKey`, and adding a `@property` for `privateKey`.

Or, if you don't feel like doing that, you can remove the `__hash__` implementation.

Reply:
Yeah, I was looking into doing the `_privateKey` approach for implicit private scope, but wasn't sure if implementing something like a getter was the approach we'd want as I'm assuming there are implementations utilizing `.privateKey` and I wanted the change to be passive. I've dabbled with Python, but not super extensively, so still finding my way around. Will look into the `@property` annotation. :)

Reply:
Ah, great. It's definitely worth a look. Something like:

```py
@property
def privateKey(self):
    return self._privateKey
```

That approach, by default, will prevent any callers from modifying the account with `acct.privateKey = some_new_key`, but still give access to use `extracted_key = acct.privateKey`.